python/pyspark/sql/utils.py

```diff
@@ -210,3 +210,14 @@ def to_str(value):
         return value
     else:
         return str(value)
+
+
+def is_timestamp_ntz_preferred():
```

Comment:
shouldn't this be a method of `SparkSession` in pyspark?

Reply:
We can. I took this out of `SparkSession` because
1. there are some places like `types.py` where some methods can be accessed without initiating `SparkSession`
2. it doesn't require any properties in `SparkSession` instance
3. some tests fail for `SQLContext` (e.g., https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/143016/console)
4. When `SparkContext` or `SparkSession` is stopped, JVM is still supposed to be alive in PySpark. It's safer to don't assume on `SparkSession`.